internal/liqonet/network-manager/tunnelendpointcreator/tunnelendpointcreator.go

```diff
@@ -256,11 +256,17 @@ func (tec *TunnelEndpointCreator) enforceRemoteNetConfigMeta(ctx context.Context
 	// to be triggered and reconcile its status when this NetworkConfig changes.
 	if utils.GetOwnerByKind(&netcfg.OwnerReferences, "ForeignCluster") == nil {
 		fc, err := foreignclusterutils.GetForeignClusterByID(ctx, tec.Client, clusterID)
-		if err != nil {
+		if client.IgnoreNotFound(err) != nil {
 			klog.Errorf("Failed to retrieve ForeignCluster associated with NetworkConfig %q: %v", klog.KObj(netcfg), err)
 			return err
 		}
 
+		// It could happen that the networkconfig is created before the foreigncluster.
+		if err != nil {
+			klog.V(4).Infof("Failed to retrieve ForeignCluster associated with NetworkConfig %q: %v", klog.KObj(netcfg), err)
+			return nil
+		}
```

Comment:
I am missing why this is needed, as the final outcome seems the same to me.

Reply:
If we need to set up the networking before the peering is enabled, i.e foreigncluster does not exist we have to process the networkconfigs even if the fc is not found. So I added for debugging purposes a the log when the fc is not found.